plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -2051,20 +2051,52 @@ instance uni ~ DefaultUni => ToBuiltinMeaning uni DefaultFun where
             (runCostingFunSixArguments . unimplementedCostingFun)
 
     toBuiltinMeaning _semvar DropList =
-        let dropListDenotation :: Integer -> SomeConstant uni [a] -> BuiltinResult (Opaque val [a])
+        let dropListDenotation
+                :: IntegerCostedLiterally -> SomeConstant uni [a] -> BuiltinResult (Opaque val [a])
             dropListDenotation i (SomeConstant (Some (ValueOf uniListA xs))) = do
                 -- See Note [Operational vs structural errors within builtins].
                 case uniListA of
                     DefaultUniList _ ->
+-- We only support @base-4.15@ and higher, because prior versions don't expose the same interface to
+-- 'Integer'.
 #if MIN_VERSION_base(4,15,0)
-                        fromValueOf uniListA <$> case i of
+                        -- The fastest way of dropping elements from a list is by operating on
+                        -- an unboxed int (i.e. an 'Int#'). We could implement that manually, but
+                        -- 'drop' in @Prelude@ already does that under the hood, so we just need to
+                        -- convert the given 'Integer' to an 'Int' and call 'drop' over that.
+                        fromValueOf uniListA <$> case unIntegerCostedLiterally i of
```

Comment:
Is there any reason to prefer `unIntegerCostedLiterally` here rather  than doing 
````
dropListDenotation (IntegerCostedLiterally i) (SomeConstant (Some (ValueOf uniListA xs))) = ...
```
?

I'm not complaining, just wondering if there's some convention or something.  Presumably neither of these incurs any computational cost.

Reply:
I initially wrote the former, but it resulted in this clunky long line, so I just turned it into the latter. It doesn't matter either way.